homeassistant/components/xiaomi_tv/media_player.py

```diff
@@ -14,50 +15,56 @@
 import homeassistant.helpers.config_validation as cv
 
 DEFAULT_NAME = "Xiaomi TV"
+CONF_ASSUME_STATE = "assume_state"
 
 _LOGGER = logging.getLogger(__name__)
 
-SUPPORT_XIAOMI_TV = SUPPORT_VOLUME_STEP | SUPPORT_TURN_ON | SUPPORT_TURN_OFF
+SUPPORT_XIAOMI_TV = (
+    SUPPORT_VOLUME_STEP | SUPPORT_TURN_ON | SUPPORT_TURN_OFF | SUPPORT_SELECT_SOURCE
+)
 
 # No host is needed for configuration, however it can be set.
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
     {
         vol.Optional(CONF_HOST): cv.string,
         vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        vol.Optional(CONF_ASSUME_STATE, default=True): cv.boolean,
     }
 )
 
 
-def setup_platform(hass, config, add_entities, discovery_info=None):
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
```

Comment:
The library doesn't support asyncio. Please revert to `setup_platform`.

Reply:
I will try to rewrite the pymitv to use aiohttp instead of requests to support asyncio. But for now I revert this commit.